sensu/plugins/check-dir-new-files.rb

```diff
@@ -0,0 +1,98 @@
+#! /usr/bin/env ruby
+#
+#   check-dir-new-files
+#
+# DESCRIPTION:
+#   Checks the number of specific files in a directory
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux, BSD
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#   #YELLOW
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2014 Sonian, Inc. and contributors. <support@sensuapp.org>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'fileutils'
+require 'time'
+
+class DirCount < Sensu::Plugin::Check::CLI
+  BASE_DIR = '/var/cache/check-dir-new-files'
+  
+  option :directory,
+         description: 'Directory to count files in',
+         short: '-d DIR',
+         long: '--dir DIR',
+		 default: '/var/crash'
+
+  option :filename_pattern,
+         description: 'filename patten to match',
+         short: '-p PATTERN',
+         long: '--pattern PATTERN',
+		 default: '*.crash'
+
+  option :criticality,
+         description: "Set sensu alert level, default is critical",
+         short: '-z CRITICALITY',
+         long: '--criticality CRITICALITY',
+         default: 'critical'
+
+  def getLastCheckTime()
+    begin
+        @state_file = File.join(BASE_DIR,config[:directory])
```

Comment:
maybe we can swap the slashes for underscores in the directory name here ?  that way we don't end up with a really deep directory structure.   also do we need to flock the file so that we don't get multiple checks fighting over it ?